thrust/thrust/system/cuda/detail/sort.h

```diff
@@ -299,7 +299,14 @@ namespace __smart_sort
 template <class Key, class CompareOp>
 using can_use_primitive_sort = ::cuda::std::integral_constant<
   bool,
-  ::cuda::std::is_arithmetic<Key>::value
+  (::cuda::std::is_arithmetic<Key>::value
+#  ifdef _CCCL_HAS_HALF
+   || ::cuda::std::is_same<Key, __half>::value
+#  endif
+#  ifdef _CCCL_HAS_NVFP16
```

Comment:
critical: `_CCCL_HAS_NVFP16` indicates presense of `__half` not bf16. To detect bf16 we have `_CCCL_HAS_NVBF16`:
```suggestion
#  ifdef _CCCL_HAS_NVBF16
```

Reply:
Which reminds me that we don't have proper header testing in Thrust. @bernhardmgruber could you add something along the following lines to Thrust?

https://github.com/NVIDIA/cccl/blob/24ed47daab316859c84652b19f563b2f355f0241/cub/cmake/CubHeaderTesting.cmake#L48

Reply:
Fixed the macro name. For the header testing, could you please open a separate issue please? Thx!

Reply:
@bernhardmgruber I don't think this can be done as a separate PR. Thrust didn't use to reference extended floating point types. After your PR it will. Without a test, it's impossible to know if this PR breaks users relying on, say `CCCL_DISABLE_BF16_SUPPORT` or not. The fact that we almost leaked this breakage is indicator of this issue.

Reply:
PR opened: #2219

Reply:
PR merged.

Reply:
> I don't think this can be done as a separate PR. 

My bad, I completely missed the idea of merging separate PR before this one. @bernhardmgruber thank you for going out of your way to make a small PR! This should improve review times and keep the PR scope small. I’m learning a lot from your approach, and we should definitely do this more often moving forward.